### PR TITLE
Add tutorials from ARM DevSummit hands-on workshop

### PR DESCRIPTION
**Pull request by ghost**
These tutorials were originally created for the Autoware workshop that was given during the ARM DevSummit in October 2020.
Since the original hands-on exercises used Autoware Online, I've modified the exercises accordingly, but I'm sure there is still room for improvement.

If this PR is approved, please note that there is an additional zipped rosbag that needs to be uploaded somewhere for the second tutorial on localization (possibly the same Google Drive folder where the other sample maps and rosbag files are stored?), and then the 2_localization.md file will need to be updated.

### CODE REVIEW

**Comment by kmiya** [comment]
Awesome! :tada: 
IMHO adding links to the tutorials to [README.md](https://github.com/tier4/AutowareArchitectureProposal.proj/blob/master/README.md) would be more helpful for Autoware users.

**Comment by ghost** [comment]
@kmiya Thank you for the comment! I've updated the README.md file as suggested.

**Comment by UMiho** [comment]
Additional tutorials look great!

BTW, how about merging /docs/SimulationTutorial.md into /docs/tutorials and reducing a redundancy of the contents?
If new tutorials are added, there will be 3 similar explanations (in /README.md, /docs/SimulationTutorial.md, /docs/tutorials/*.md) and it may cause an inconsistency of the contents when we update them in future.

In fact, there are already some diffs between 3 pages as far as I took a glance.

e.g.1) rosbag simulation launch command
- `roslaunch autoware_launch logging_simulator.launch map_path:=/path/to/map_folder vehicle_model:=lexus sensor_model:=aip_xx1 rosbag:=true` in /README.md
- `roslaunch autoware_launch logging_simulator.launch map_path:=/path/to/map_folder` in /docs/SimulationTutorial.md
- `roslaunch autoware_launch logging_simulator.launch vehicle_model:=lexus sensor_model:=aip_xx1 map_path:=/path/to/tutorial1_maps` in /docs/tutorials/1_rosbag.md

e.g.2) URL of the map download folder
- `https://drive.google.com/open?id=1ovrJcFS5CZ2H51D8xVWNtEvj_oiXW-zk` in /README.md and /docs/tutorials/1_rosbag.md
- `https://drive.google.com/open?id=197kgRfSomZzaSbRrjWTx614le2qN-oxx` in /docs/SimulationTutorial.md

File: docs/tutorials/1_rosbag.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+Before starting, please download and unpack the following files:
+* [Tutorial 1 map files](https://drive.google.com/open?id=1ovrJcFS5CZ2H51D8xVWNtEvj_oiXW-zk)
+* [Tutorial 1 rosbag](https://drive.google.com/open?id=1BFcNjIBUVKwupPByATYczv2X4qZtdAeD)
+
+> Note that these are the same files used in the [rosbag quick launch exercise](../../README.md#rosbag-simulation)
+
+## Launch RViz, start and pause rosbag playback
+1. Open two terminal windows and run the following commands in *both* terminals
+```
+cd ~/workspace/AutowareArchitectureProposal
```
Comment (UMiho) [inline docs/tutorials/1_rosbag.md]: `~/workspace/AutowareArchitectureProposal` is `~/workspace/AutowareArchitectureProposal.proj` now.
This change should be reflected to all command examples in this repository.

Comment (kmiya) [reply docs/tutorials/1_rosbag.md]: @UMiho 470b6e0ac675506e1575e5b6f98dc0f2a3be5b80 fixes the issue!

**Comment by kmiya** [comment]
@LalithVipulananthan @UMiho Commit 5287b92a5c91d1e3a2d78e6f9eb6a9c7b8568038 applies some [markdownlint](https://github.com/DavidAnson/markdownlint) rules, removes extra spaces, fixes a broken image link, etc. It would be appreciated if you could check them.

**Comment by ghost** [comment]
> BTW, how about merging /docs/SimulationTutorial.md into /docs/tutorials and reducing a redundancy of the contents?
> If new tutorials are added, there will be 3 similar explanations (in /README.md, /docs/SimulationTutorial.md, /docs/tutorials/*.md) and it may cause an inconsistency of the contents when we update them in future.

@UMiho If you think it will be okay to merge the contents, then I'm happy to revise the 3_simulation.md file to include additional information from /docs/SimulationTutorial.md.

Do you or @kmiya know where I can upload the additional rosbag file that is needed for the Localization tutorial?

**Comment by kmiya** [comment]
> Do you or @kmiya know where I can upload the additional rosbag file that is needed for the Localization tutorial?

I think Google Drive of the `public.tier4.jp` account might be the place to upload the rosbag. Please ask ICT team on Slack.

**Comment by UMiho** [comment]
> @UMiho If you think it will be okay to merge the contents, then I'm happy to revise the 3_simulation.md file to include additional information from /docs/SimulationTutorial.md.

@LalithVipulananthan I think it would be better that splitting information in /docs/SimulationTutorial.md to 0_preparation.md and 1_rosbag.md because SimulationTutorial.md includes not only planning simulation info but also rosbag simulation.

My idea is as below:
- move following part to 0_preparation.md
```
Autoware provides two types of simulation:
- rosbag-based simulation that can be used for testing/validation of the `Sensing`, `Localization` and `Perception` stacks.
- The Planning Simulator tool which is mainly used for testing/validation of `Planning` stack by simulating traffic rules, interactions with dynamic objects and control commands to the ego vehicle.
![sim](https://user-images.githubusercontent.com/8327598/79709776-0bd47b00-82fe-11ea-872e-d94ef25bc3bf.png)
```
- move a table below to 1_rosbag.md
```
| Sensor                | Topic name                               |
| --------------------- | ---------------------------------------- |
| Velodyne 128 (Top)    | /sensing/velodyne/top/velodyne_packets   |
| Velodyne 16 (Right)   | /sensing/velodyne/right/velodyne_packets |
| Velodyne 16 (Left)    | /sensing/velodyne/left/velodyne_packets  |
| IMU (Tamagawa TAG300) | /sensing/imu/tamagawa/imu_raw            |
| GNSS (Ublox F9P)      | /sensing/gnss/ublox/fix_velocity         |
|                       | /sensing/gnss/ublox/nav_sat_fix          |
|                       | /sensing/gnss/ublox/navpvt               |
| CAN data              | /vehicle/status/control_mode             |
|                       | /vehicle/status/shift                    |
|                       | /vehicle/status/steering                 |
|                       | /vehicle/status/twist                    |
| ~~Camera x 7~~        | ~~/sensing/camera/camera[]/image_raw~~   |
> Note: Due to privacy concerns, image data has been removed from the rosbag file.
```
- ignore the rest of /docs/SimulationTutorial.md because it's already covered in 1_rosbag.md and 3_simulation.md

What do you think about this?

**Comment by ghost** [comment]
@UMiho @kmiya Apologies for the extensive changes. I merged the SimulationTutorial.md contents as suggested, but I also added details of how to simulate a parking manoeuvre to 3_simulation.md, which required some additional screenshot images. 

I also added the images used in SimulationTutorial.md to the tutorial images folder, updated the link to the new rosbag file, updated the README.md file and lastly removed the SimulationTutorial.md file.

**Comment by ghost** [comment]
@yukkysaito (@kmiya) According to ICT, sharing files using a `public.tier4.jp` account requires making a purchase request to have a new `public.tier4.jp` account created. Alternatively, ICT suggested asking someone who already has such an account.

Saito-san, my understanding is that you have a `public.tier.jp` account. If so, would it be possible to share the following file please?
https://drive.google.com/file/d/1fEkNe0IksR9g_lq8aypZaGKxWtyaSZ2T

File: docs/tutorials/0_preparation.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+- [Tutorial 2: Localization using NDT Scan Matching](2_localization.md)
+- [Tutorial 3: Simulation testing with dummy objects](3_simulation.md)
+
+We assume that you've [installed and set up the AutowareArchitectureProposal already](https://github.com/tier4/AutowareArchitectureProposal.proj#autoware-setup).
+
+## Simulation in Autoware
+
+Autoware provides two types of simulation:
+- rosbag-based simulation that can be used for testing/validation of the `Sensing`, `Localization` and `Perception` stacks (this was covered in the first two tutorials)
+- The Planning Simulator tool which is mainly used for testing/validation of `Planning` stack by simulating traffic rules, interactions with dynamic objects and control commands to the ego vehicle (covered in this tutorial)
```
Comment (UMiho) [inline docs/tutorials/0_preparation.md]: @LalithVipulananthan `(covered in this tutorial)` -> `(covered in the third tutorial)` ?

Comment (ghost) [reply docs/tutorials/0_preparation.md]: Thank you! I've updated the file accordingly.

**Review by UMiho — APPROVED** [review]
Finally, it looks good to me!

**Review by kmiya — APPROVED** [review]
LGTM except the following issue.

> Saito-san, my understanding is that you have a public.tier.jp account. If so, would it be possible to share the following file please?
https://drive.google.com/file/d/1fEkNe0IksR9g_lq8aypZaGKxWtyaSZ2T

**Comment by kmiya** [comment]
@LalithVipulananthan 
Hi. Has the account of the `public.tier4.jp` already been available for you now?

**Comment by ghost** [comment]
> @LalithVipulananthan
> Hi. Has the account of the `public.tier4.jp` already been available for you now?

My purchase request for a public.tier4.jp account has been approved. I'm just waiting for IT to set up the account for me. Once that is done, I will upload the rosbag file, update the tutorial file and notify you here.

**Comment by ghost** [comment]
> @LalithVipulananthan
> Hi. Has the account of the `public.tier4.jp` already been available for you now?

@kmiya Apologies for the delay. I just set up my public.tier4.jp account today, uploaded the rosbag minus GNSS file and updated the link in the second tutorial. The PR should be good to go now! :)

**Comment by kmiya** [comment]
@LalithVipulananthan Great :+1: Thank you for everything!

@UMiho Now this PR is ready to merge!

**Pull request merged**